src/mjml/mjml.service.js

```diff
@@ -43,21 +43,42 @@ export default class MjmlService {
 
   /**
    * Transform MJML to HTML
-   * @todo show validation erros in the UI
+   * @todo show validation errors in the UI
    * @returns string
    */
-  static mjmlToHtml(mjml) {
+  static mjmlToHtml(mjml, endpoint = '') {
+    let html = '';
+
     try {
       if (typeof mjml !== 'string' || !mjml.includes('<mjml>')) {
         throw new Error('No valid MJML provided');
       }
-      // html needs to be beautified for the click tracking to work.
-      // strict mode not working with e.g. id="" and data-type parameters that
-      // are e.g. used for Dynamic Content
-      return mjml2html(mjml, { beautify: true });
+
+      if (endpoint !== '') {
+        html = MjmlService.mjmlToHtmlViaEndpoint(mjml, endpoint);
+      } else {
+        // html needs to be beautified for the click tracking to work.
+        // strict mode not working with e.g. id="" and data-type parameters that
+        // are e.g. used for Dynamic Content
+        html = mjml2html(mjml, { beautify: true });
+      }
     } catch (error) {
       console.warn(error);
-      return null;
     }
+
+    return html;
+  }
+
+  /**
+   * Transform MJML to HTML via endpoint
+   * @returns string|null
+   */
+  static mjmlToHtmlViaEndpoint(mjml, endpoint) {
+    const xmlHttpRequest = new XMLHttpRequest();
+    xmlHttpRequest.open('POST', endpoint, false);
+    xmlHttpRequest.setRequestHeader('Content-type', 'application/json');
+    xmlHttpRequest.send(JSON.stringify({ mjml }));
+
+    return xmlHttpRequest.responseText ? JSON.parse(xmlHttpRequest.responseText) : null;
```

Comment:
@volha-pivavarchyk  same here.. :) you can simplify it and always return a string. No null